gobblin-restli/gobblin-throttling-service/gobblin-throttling-service-server/src/main/java/gobblin/restli/throttling/ThrottlingGuiceServletConfig.java

```diff
@@ -52,34 +62,56 @@
  */
 @Slf4j
 @Getter
-public class ThrottlingGuiceServletConfig extends GuiceServletContextListener {
+public class ThrottlingGuiceServletConfig extends GuiceServletContextListener implements Closeable {
 
-  private ServletContext _context;
-  private Config _brokerConfig;
+  public static final String THROTTLING_SERVER_PREFIX = "throttlingServer.";
+  public static final String LISTENING_PORT = THROTTLING_SERVER_PREFIX + "listeningPort";
+  public static final String HOSTNAME = THROTTLING_SERVER_PREFIX + "hostname";
+  public static final String ZK_STRING_KEY = "throttling.helix.zkString";
+
+  private Optional<LeaderFinder<URIMetadata>> _leaderFinder;
+  private Config _config;
+  private Injector _injector;
 
   @Override
   public void contextInitialized(ServletContextEvent servletContextEvent) {
-    _context = servletContextEvent.getServletContext();
+    ServletContext context = servletContextEvent.getServletContext();
 
-    Enumeration<String> parameters = _context.getInitParameterNames();
-    Map<String, String> brokerConfigMap = Maps.newHashMap();
+    Enumeration<String> parameters = context.getInitParameterNames();
+    Map<String, String> configMap = Maps.newHashMap();
     while (parameters.hasMoreElements()) {
       String key = parameters.nextElement();
-      if (key.startsWith(BrokerConstants.GOBBLIN_BROKER_CONFIG_PREFIX)) {
-        brokerConfigMap.put(key, _context.getInitParameter(key));
-      }
+      configMap.put(key, context.getInitParameter(key));
     }
-    this._brokerConfig = ConfigFactory.parseMap(brokerConfigMap);
+    initialize(ConfigFactory.parseMap(configMap));
 
     super.contextInitialized(servletContextEvent);
   }
 
+  public void initialize(Config config) {
+    try {
+      this._config = config;
+      this._leaderFinder = getLeaderFinder(this._config);
+      if (this._leaderFinder.isPresent()) {
+        this._leaderFinder.get().startAsync();
+        this._leaderFinder.get().awaitRunning(100, TimeUnit.SECONDS);
+      }
+      this._injector = createInjector(this._config, this._leaderFinder);
+    } catch (URISyntaxException | IOException | TimeoutException exc) {
+      log.error(String.format("Error in %s initialization.", ThrottlingGuiceServletConfig.class.getSimpleName()), exc);
+      throw new RuntimeException(exc);
+    }
+  }
+
   @Override
-  protected Injector getInjector() {
-    return getInjector(this._brokerConfig);
+  public Injector getInjector() {
+    return this._injector;
   }
 
-  public static Injector getInjector(final Config brokerConfig) {
+  private Injector createInjector(final Config config, final Optional<LeaderFinder<URIMetadata>> helixManager) {
```

Comment:
Since helix is not used, maybe change all references to Helix to a more generic name.